nginx-controller/controller/utils.go

```diff
@@ -175,7 +175,7 @@ func (s *StoreToIngressLister) List() (ing extensions.IngressList, err error) {
 // Note that this ignores services without the right nodePorts.
 func (s *StoreToIngressLister) GetServiceIngress(svc *api_v1.Service) (ings []extensions.Ingress, err error) {
 	for _, m := range s.Store.List() {
-		ing := *m.(*extensions.Ingress)
+		ing := *m.(*extensions.Ingress).DeepCopy()
```

Comment:
why?

Reply:
When endpoints change we fetch ingresses using this method (via `lbc.getIngressForEndpoints`) in `syncEndp` which calls the same `UpdateEndpointsMergeableIngress` that messes with the annotations.